include/graphblas/hyperdags/io.hpp

```diff
@@ -332,6 +332,38 @@ namespace grb {
 		return ret;
 	}
 
+	template<
+		Descriptor descr = descriptors::no_operation,
+		typename DataType, typename RIT, typename CIT, typename NIT,
+		typename ValueType = DataType
+	>
+	RC set(
+		Matrix< DataType, hyperdags, RIT, CIT, NIT  > &C,
+		const ValueType& val,
+		const Phase &phase = EXECUTE
+	) noexcept {
+		const RC ret = set< descr >(
+			internal::getMatrix( C ), val, phase
+		);
+		if( ret != SUCCESS ) { return ret; }
+		if( phase != EXECUTE ) { return ret; }
+		if( nrows( C ) == 0 || ncols( C ) == 0 ) { return ret; }
+		std::array< const void *, 0 > sourcesP{};
+		std::array< uintptr_t, 1 > sourcesC{
+			getID( internal::getMatrix(C) )
+		};
+		std::array< uintptr_t, 1 > destinations{
```

Comment:
I think also val should be registered as a source scalar. There should be multiple examples of that already in the hyperdags backend, for example, for folding a scalar into a vector (foldl/foldr)